src/App.tsx

```diff
@@ -1,61 +1,133 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import TodoList from './components/TodoList/TodoList';
+
+const initialData = () => {
+  return (
+    todosFromServer.map((todo) => {
+      return ({
+        ...todo,
+        user: usersFromServer
+          .find((user: User) => user.id === todo.userId) || null,
+      });
+    })
+  );
+};
 
 export const App = () => {
+  const [titleError, setTitleError] = useState(false);
+  const [selectError, setSelectError] = useState(false);
+  const [newTitle, setNewTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('');
+  const [todosForRender, setTodos] = useState(initialData);
+
+  const todoUpdater = () => {
+    const maxId = Math.max(...todosForRender.map(({ id }) => id), 0);
+
+    const newTodo: Todo = {
+      id: (maxId + 1),
+      title: newTitle,
+      completed: false,
+      userId: +selectedUserId,
+      user: usersFromServer
+        .find((user: User) => user.id === +selectedUserId) || null,
+    };
+
+    const updatedTodos = todosForRender.map(todo => ({ ...todo }));
+
+    updatedTodos.push(newTodo);
+
+    setTodos(updatedTodos);
```

Comment:
use prevState, no need to do map then push new value
```suggestion
    setTodos(prevTodos => [...prevTodos, newTodo]);
```